terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -81,6 +91,42 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
+      def update_lockfile_declaration(dependency)
+        new_req = dependency.requirements.first
+        lockfile = lock_file.first
+        content = lockfile.content.dup
+
+        provider_source = new_req[:source][:registry_hostname] + "/" + new_req[:source][:module_identifier]
+        declaration_regex = lockfile_declaration_regex(provider_source)
+        lockfile_dependency_removed = content.sub(declaration_regex, "")
+
+        copy_dir_to_temporary_directory do
```

Comment:
This is a little nitty, but I think this would be slightly more idiomatic for the codebase:

```ruby
SharedHelpers.in_a_temporary_directory do
  write_dependency_files

  File.write(".terraform.lock.hcl", lockfile_dependency_removed)
  # ...

def write_dependency_files
  dependency_files.each do |file|
    # Do not include the .terraform directory or .terraform.lock.hcl
    next if file.name.include?(".terraform")

    File.write(file.name, file.content)
```